### DISCUSSION TITLE
FDO Profiles - Kernel Attributes

### DISCUSSION THREAD

**Discussion by juliancervos** [BioDT/biodt-fair#3] [General]
The kernel attributes are mandatory to all BioDT FDOs regardless of their type. They contain fundamental information that any FAIR Digital Objects (FDOs) should have as part of their metadata description (such as the type of digital object, license, author or the creation date...). 

In the table below, a basic set of kernel attributes has been drafted. For now, they have been kept to a minimum to ease adoption by the different prototype Digital Twins (pDTs). As more attributes are needed, they can be incorporated in the table below.

Table format: the name of the metadata attributes is shown in the "Property" column, including a link to [Schema.org](https://schema.org/) (or another metadata schema from which the property was taken), and a definition in the "Description" column. The "Type" column specifies the expected values for each property, while the "Cardinality" column specifies the amount and whether they are optional or mandatory. Lastly, additional remarks might be added to the "Comments" column, and an example is given in the final column.

## Kernel Attributes Table

| Property | Description | Type | Cardinality | Comments | Example |
| :--: | :--- | :--: | :--: | :--- | :--: |
| [profile](https://www.w3.org/ns/dx/prof/Profile) | The FDO profile that a certain metadata description follows. | [URL](https://schema.org/URL) | 1/1 | For now, we can refer to this repo for the profiles. | `"profile": "https://github.com/BioDT/biodt-fair/discussions/4"` |
| [license](http://schema.org/license) | License for the FDO the metadata description is about. | [URL](https://schema.org/URL) | 1/1 | The FDO might have a certain license, however, the metadata file should have a CC0 license. | `"license": {"@id": "https://creativecommons.org/licenses/by/4.0/"}` |
| [@type](https://www.w3.org/TR/json-ld/#specifying-the-type) | The nature of the object the metadata describes. In BioDT, we currently consider "Model", "Dataset" and "Workflow". | [Text](https://schema.org/Text) | 1/many | `"@type"` is a reserved term in JSON-LD, and there are certain restrictions to its usage. Here, apart from complying with JSON-LD and RO-Crate, we use that attribute to determine the digital object type. | `"@type": ["File", "SoftwareSourceCode", "ComputationalWorkflow"]` |
| [@id](https://www.w3.org/TR/json-ld/#node-identifiers) | An identifier for the FDO. Ideally, a globally unique, persistent and resolvable identifier. | [IRI](https://datatracker.ietf.org/doc/html/rfc3987#section-2) | 1/1 | As an alternative for `"@id"` (for example, for other serialisations), [`"identifier"`](http://schema.org/identifier) can be used in addition to it in order to specify a PID. | `"@id": "https://doi.org/10.1111/1365-2664.12222"` |
| [name](https://schema.org/name) | Name of the FDO. | [Text](https://schema.org/Text) | 1/1 |  | `"name": "BEEHAVE model"` |
| [description](http://schema.org/description) | Short description of what the FDO is. | [Text](https://schema.org/Text) | 1/1 |  | `"description": "BEEHAVE is a computer model to simulate the development of a honeybee colony and its nectar and pollen foraging behavior in different..."` |
| [dateCreated](http://schema.org/dateCreated) | Date of creation of the FDO. | [Date](http://schema.org/Date)  <br>[DateTime](http://schema.org/DateTime) | 1/1 | `"datePublished"` and `"dateModified"` can also be included to add further details. | `"dateCreated": "2020-04-07 13:16:09 UTC"` |
| [author](http://schema.org/author) | Author of the FDO. | [Organization](http://schema.org/Organization)  <br>[Person](http://schema.org/Person) | 1/many | If possible, use an ORCiD or ROR ID, otherwise use full name in natural order. | `"author": [{"@id": "http://orcid.org/0000-0003-0791-7164"}, {"@id": "http://orcid.org/0000-0002-3221-9512"}, {"@id": "http://orcid.org/0000-0003-2338-4636"}],` |
| [keywords](https://schema.org/keywords) | A comma-separated list of keywords or tags used to describe some item. | [Text](https://schema.org/Text) | 1/many |  | `"keywords": ["bioacoustics", "bird sound recognition", "deep learning"]` |

Here, we can discuss different aspects of the kernel attributes. For example:
- Is there any important information that should be included as a kernel attribute and is currently missing?

**Comment by stain** [comment]
For profile reference then https://www.w3.org/TR/dx-prof/ [uses](https://www.w3.org/TR/dx-prof/#Property:conformsTo dct:conformsTo](http://purl.org/dc/terms/conformsTo) to a [prof:Profile](https://www.w3.org/TR/dx-prof/#Class:Profile) resource. This is also how we use it in [RO-Crate profile conformance](https://www.researchobject.org/ro-crate/1.2-DRAFT/profiles#declaring-conformance-of-an-ro-crate-profile) where the object may be another RO-Crate (a Profile Crate) that is also a `prof:Profile`.
